torch/_dynamo/output_graph.py

```diff
@@ -244,9 +258,10 @@ def current_tx(self):
     def copy_graphstate(self) -> OutputGraphState:
         """Create a checkpoint of the current state by copying everything"""
         assert self.nn_modules is not None
+        guards_graph_state = self.tracing_context.guards_context.copy_graphstate()
         state = OutputGraphState(
             list(self.graphargs),
-            set(self.guards),
+            guards_graph_state,
```

Comment:
The "ownership" (who is responsible for calling checkpoint on the tracing context) is a bit strange here. Does Dynamo output graph "own" the tracing context? But I'm not sure how to do it better... should we call checkpoint on the TracingContext only? How does that work... mumble mumble callback mumble. (No action needed.)